lib/tasks/annual_fund_impact_metrics_activities.rake

```diff
@@ -0,0 +1,43 @@
+require "csv-safe"
+
+namespace :activities do
+  task :annual_fund_impact_metrics, [:output_csv_path] => :environment do |_task, args|
+    desc "Generates a CSV file of Activities that will be used to pre-populate an annual
+          collaborative spreadsheet of fund impact metrics"
+
+    csv_path = args[:output_csv_path] || "tmp/activities.csv"
+
+    CSVSafe.open(csv_path, "wb") do |csv|
+      activities = Activity
+        .joins(:organisation)
+        .includes(:organisation, :actuals)
+        .where.not(programme_status: ["delivery", "agreement_in_place", "open_for_applications"])
+        .order("organisations.name, programme_status")
+
+      # We have decided that the most accurate way to determine whether an Activity
+      # was completed or stopped more than 2 years ago, is to exclude those that have not had
+      # any Actuals reported in the last two years.
+      active_activities = activities.reject do |activity|
+        completed_or_stopped = ["completed", "stopped"].any?(activity.programme_status)
+        has_actuals = activity.actuals.any?
+        actuals_reported_in_last_two_years = activity.actuals.any? { |actual| actual.date >= 2.years.ago.to_date }
+
+        completed_or_stopped && has_actuals && !actuals_reported_in_last_two_years
```

Comment:
To make sure I understand the logic: we're ok with not rejecting complete or stopped activities with no actuals? (Because we can't determine with certainty how long ago the activity was completed or stopped.)

Reply:
That was my own assumption. BEIS said that we should reject any Activities with no Actuals in the last two years. 

But wouldn't a brand-new Activity have no Actuals at all, and we don't want that rejected?

Reply:
There seems to be over 2000 Activities like that `Activity.count { |a| a.actuals.count == 0 && (a.programme_status == "completed" || a.programme_status == "stopped") }`

Reply:
Thanks for spotting this, the assumption I made was definitely wrong!

Reply:
TIL you can pass a block to `.count`